content/learn/intermediate/outputs-2.md

```diff
@@ -0,0 +1,249 @@
+creating a new one each time something changes. We provide the `editMessage` function for this purpose.
+
+```go
+{{ editMessage channel_id message_id new_message_content }}
+```
+
+The `channel_id` is the ID of the channel in which the message is located. The `message_id` is the ID of the message you
+want to edit. The `new_message_content` is the new content you want to set for the message, completely replacing it.
+You should note that YAGPDB can only edit messages from itself, just like you cannot edit someone else's messages.
+
+For a quick demonstration, consider the following code:
+
+```go
+{{ $messageID := sendMessageRetID nil "Hello, World!" }}
+{{ sleep 5 }}
+{{ editMessage nil $messageID "Goodbye, World!" }}
+```
+
+This code sends a message saying "Hello, World!" and then, after 5 seconds, edits the message to say "Goodbye, World!",
+all in the same channel where the custom command was triggered.
+
+### Editing Embeds
+
+Editing embeds is a little more involved than editing regular messages. You need to provide the entire embed object
+again, old parts included. This is due to how the Discord API functions. The basic idea is that we obtain the old embed
+object, modify it, and then send the modified object back to the `editMessage` function.
+
+A contrived example all within the same custom command looks like the following:
+
+```go
+{{ $embed := cembed
+    "title" "This is a title"
+    "description" "This is a description."
+    "color" 0xff0000
+    "fields" (cslice
+        (sdict "name" "Field 1" "value" "Value 1" "inline" true)
+        (sdict "name" "Field 2" "value" "Value 2" "inline" true)
+    )
+    "author" (sdict
+      "name" .User.Username
+      "icon_url" (.User.AvatarURL "256")
+    )
+}}
+{{ $messageID := sendMessageRetID nil $embed }}
+{{ sleep 5 }}
+
+{{ $new_embed := structToSdict $embed }}
+{{ $new_embed.Set "title" "This is a new title" }}
+{{ $new_embed.Set "description" "This is a new description." }}
+{{ $new_embed.Set "fields" (cslice) }}
+{{ editMessage nil $messageID (cembed $new_embed) }}
+```
+
+In the second part, after `{{ sleep 5 }}`, we first convert the original embed object to a dictionary using the
+`structToSdict` function. We then modify the fields we want to change using the `Set` method. In this case, we change
+the title and description of the embed, and remove all fields. Finally, we send the modified embed object back to the
+`editMessage` function.
+
+As mentioned previously, this example is contrived and meant to demonstrate the process. In practice, we're more likely
+to use `.Message.Embeds` to get the original embed object, convert and modify it as shown above, then send it back to
+the `editMessage` function.
+
+{{< callout context="note" title="Note" icon="outline/info-circle" >}}
+
+`structToSdict` does not perform deep conversion and has some other general inconsistencies. For a full conversion of an
```

Comment:
`structToSdict` does not perform deep conversion and does not convert the case of field names.

(structToSdict isn't inconsistent; it's perfectly well-defined--but it isn't sufficient for the use-case.)

Reply:
Ah, that was me misinterpreting the `embedtosdict` snippet from the support server, there it says:

```
tructToEmbed does not change the keys so there are some inconsistencies (causing some parts of the embed to be left out without the following code).
```

Certainly well-defined, as you said, just initially unintuitive.